runtime/tests/port/j9strTest.c

```diff
@@ -1604,35 +1609,71 @@ j9str_test_atoe_vsnprintf(struct J9PortLibrary *portLibrary)
 
 	/* min_width is less than the length of inputString */
 	j9tty_printf(PORTLIB, "\n\tTesting case 2\n");
-	rc |= printToBuffer(portLibrary, FALSE, buffer2, bufferLength2, expectedOutput2, "%*s", 2, nonNullTerminatedString);
+	rc |= printToBuffer(portLibrary, FALSE, buffer2, bufferLength2, expectedOutput2, strlen(nullTerminatedString), "%*s", 2, nullTerminatedString);
 
 	/* min_width is equal to the length of inputString (buffer length > min_width) */
 	j9tty_printf(PORTLIB, "\n\tTesting case 3\n");
-	rc |= printToBuffer(portLibrary, FALSE, buffer1, bufferLength1, expectedOutput1, "%*s", 4, nullTerminatedString);
+	rc |= printToBuffer(portLibrary, FALSE, buffer1, bufferLength1, expectedOutput1, strlen(expectedOutput1), "%*s", 4, nullTerminatedString);
 
 	/* min_width is greater than the length of inputString (buffer length > min_width) */
 	j9tty_printf(PORTLIB, "\n\tTesting case 4\n");
-	rc |= printToBuffer(portLibrary, FALSE, buffer3, bufferLength3, expectedOutput3, "%*s", 10, nullTerminatedString);
+	rc |= printToBuffer(portLibrary, FALSE, buffer3, bufferLength3, expectedOutput3, strlen(expectedOutput3), "%*s", 10, nullTerminatedString);
 
 	/* precision is equal to the length of inputString */
 	j9tty_printf(PORTLIB, "\n\tTesting case 5\n");
-	rc |= printToBuffer(portLibrary, TRUE, buffer0, bufferLength0, expectedOutput1, "%.*s", 4, nonNullTerminatedString);
+	rc |= printToBuffer(portLibrary, TRUE, buffer0, bufferLength0, expectedOutput1, strlen(expectedOutput1), "%.*s", 4, nonNullTerminatedString);
 
 	/* precision is less than the length of inputString */
 	j9tty_printf(PORTLIB, "\n\tTesting case 6\n");
-	rc |= printToBuffer(portLibrary, TRUE, buffer0, bufferLength0, expectedOutput4, "%.*s", 2, nonNullTerminatedString);
+	rc |= printToBuffer(portLibrary, TRUE, buffer0, bufferLength0, expectedOutput4, strlen(expectedOutput4), "%.*s", 2, nonNullTerminatedString);
 
 	/* both min_width and precision are equal to the length of inputString */
 	j9tty_printf(PORTLIB, "\n\tTesting case 7\n");
-	rc |= printToBuffer(portLibrary, TRUE, buffer0, bufferLength0, expectedOutput1, "%*.*s", 4, 4, nonNullTerminatedString);
+	rc |= printToBuffer(portLibrary, TRUE, buffer0, bufferLength0, expectedOutput1, strlen(expectedOutput1), "%*.*s", 4, 4, nonNullTerminatedString);
 
 	/* the length of inputString is equal to precision but less than min_width */
 	j9tty_printf(PORTLIB, "\n\tTesting case 8\n");
-	rc |= printToBuffer(portLibrary, TRUE, buffer0, bufferLength0, expectedOutput3, "%*.*s", 10, 4, nonNullTerminatedString);
+	rc |= printToBuffer(portLibrary, TRUE, buffer0, bufferLength0, expectedOutput3, strlen(expectedOutput3), "%*.*s", 10, 4, nonNullTerminatedString);
+
+	/* the buffer is NULL. expect to estimate the length */
+	j9tty_printf(PORTLIB, "\n\tTesting case 9\n");
+	rc |= printToBuffer(portLibrary, FALSE, NULL, 0, NULL, strlen(nonNullTerminatedString), "%s", nonNullTerminatedString);
```

Comment:
`nonNullTerminatedString` is not a reasonable argument for either `strlen()` or `%s`. Please remove or adjust this test.